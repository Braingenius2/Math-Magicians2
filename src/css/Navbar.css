.navbar {
  width: 100%;
  margin-bottom: 1.5rem;
  padding: 0.8em;
  display: flex;
  justify-content: space-between;
}

.nav-menu {
  padding: 0.5em;
  display: flex;
  justify-content: space-between;
  cursor: pointer;
  list-style: none;
}

.nav-link:hover {
  color: blue;
}

.nav-menu li + li::before {
  content: " | ";
  padding: 0 0.8rem;
}

.logo {
  font-size: 2rem;
}
