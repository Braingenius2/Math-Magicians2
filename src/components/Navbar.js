import React from 'react';
import { Link } from 'react-router-dom';
import '../css/Navbar.css';

const Navbar = () => (
  <nav className="navbar">
    <h1 className="logo">Math Magicians</h1>
    <ul className="nav-menu">
      <li className="nav-link">
        <Link to="/">Home</Link>
      </li>
      <li className="nav-link">
        <Link to="/calculator">Calculator</Link>
      </li>
      <li className="nav-link">
        <Link to="/quote">Quote</Link>
      </li>
    </ul>
  </nav>
);

export default Navbar;
