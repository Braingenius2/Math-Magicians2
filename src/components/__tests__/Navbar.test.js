import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../Navbar';

describe('Navbar', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
