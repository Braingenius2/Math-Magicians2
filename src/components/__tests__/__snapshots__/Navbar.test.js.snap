// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <h1
      class="logo"
    >
      Math Magicians
    </h1>
    <ul
      class="nav-menu"
    >
      <li
        class="nav-link"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="nav-link"
      >
        <a
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li
        class="nav-link"
      >
        <a
          href="/quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
